### PR TITLE
add support for --search-filename and --terse (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
This can be useful to support bash completion of easyconfig files.

cc @damienfrancois

### CODE REVIEW

File: easybuild/tools/filetools.py [hunk 1]
```diff
@@ -367,8 +371,12 @@ def search_file(paths, query, short=False, ignore_dirs=None, silent=False):
             else:
                 hit_lines.extend([" * %s" % fn for fn in hits])
 
-    for line in var_lines + hit_lines:
-        print_msg(line, log=_log, silent=silent, prefix=False)
+    if terse:
+        for line in hits:
+            print line
```
Comment (wpoely86) [inline easybuild/tools/filetools.py]: `print()` (for py3)

File: easybuild/tools/options.py [hunk 2]
```diff
@@ -566,6 +569,10 @@ def postprocess(self):
             if not HAVE_AUTOPEP8:
                 raise EasyBuildError("Python 'autopep8' module required to reformat dumped easyconfigs as requested")
 
+        # some options imply enabling --terse
```
Comment (wpoely86) [inline easybuild/tools/options.py]: explain why in the comment because this two seem unrelated

File: easybuild/main.py [hunk 3]
```diff
@@ -223,8 +223,6 @@ def main(args=None, logfile=None, do_build=None, testing=False):
         # print location to last log file, and exit
         last_log = find_last_log(logfile) or '(none)'
         print_msg(last_log, log=_log, prefix=False)
-        cleanup(logfile, eb_tmpdir, testing, silent=True)
```
Comment (wpoely86) [inline easybuild/main.py]: so, this changes functionally? You can now combine `--last-log` with `--search`?

Comment (boegel) [inline easybuild/main.py]: hmm, that's an unintended side-effect, but does it matter?

you can indeed combine these, but I don't think it makes a lot of sense to do it

doesn't hurt, so fine?

Comment (wpoely86) [inline easybuild/main.py]: I don't know. Might bite us in the ass later?

Comment (boegel) [inline easybuild/main.py]: This should be fine, there's lots of weird combo's of options you can make now.

Using `--last-log` and `--search` together won't break anything (I checked).

**Comment by boegel** [comment]
Going in, thanks for the review @wpoely86!
